src/environment/isolator.ts

```diff
@@ -103,29 +119,31 @@ export default class Isolator {
     };
     this.componentWithDependencies = componentWithDependencies;
     this.manyComponentsWriter = new ManyComponentsWriter(concreteOpts);
-    await this.writeComponentsAndDependencies();
-    await this.installComponentPackages();
+    await this.writeComponentsAndDependencies({ keepExistingCapsule: !!opts.keepExistingCapsule });
+    await this.installComponentPackages({ skipNodeModules: !!opts.skipNodeModules });
     await this.writeLinks();
     this.capsuleBitMap = this.manyComponentsWriter.bitMap;
     return componentWithDependencies;
   }
 
-  async writeComponentsAndDependencies() {
+  async writeComponentsAndDependencies(opts = { keepExistingCapsule: false }) {
     logger.debug('ManyComponentsWriter, writeAllToIsolatedCapsule');
     this._manipulateDir();
     await this.manyComponentsWriter._populateComponentsFilesToWrite();
     await this.manyComponentsWriter._populateComponentsDependenciesToWrite();
-    await this._persistComponentsDataToCapsule();
+    await this._persistComponentsDataToCapsule({ keepExistingCapsule: !!opts.keepExistingCapsule });
   }
 
-  async installComponentPackages() {
+  async installComponentPackages(opts = { skipNodeModules: false }) {
     // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
     this.capsulePackageJson = this.componentWithDependencies.component.packageJsonFile;
     // @ts-ignore AUTO-ADDED-AFTER-MIGRATION-PLEASE-FIX!
     this.componentRootDir = this.componentWithDependencies.component.writtenPath;
     await this._addComponentsToRoot();
     logger.debug('ManyComponentsWriter, install packages on capsule');
-    await this._installWithPeerOption();
+    if (!opts.skipNodeModules) {
```

Comment:
looks like it's the same as the installNpmPackages option. (and it's just a bug it wasn't used here).
if so, let's use the existing one?